pkg/apis/eventing/v1alpha1/trigger_types.go

```diff
@@ -58,11 +58,18 @@ type TriggerSpec struct {
 	// +optional
 	DeprecatedGeneration int64 `json:"generation,omitempty"`
 
-	Broker     string          `json:"broker,omitempty"`
+	Broker string `json:"broker,omitempty"`
+
+	// +optional
+	Filter *FilterSelector `json:"filter,omitempty"`
+
 	Subscriber *SubscriberSpec `json:"subscriber,omitempty"`
+}
 
-	Type   string `json:"type,omitempty"`
-	Source string `json:"source,omitempty"`
+type FilterSelector struct {
+	Attributes map[string]string `json:"attributes,omitempty" protobuf:"bytes,1,rep,name=attributes"`
+	// TODO create our own FilterSelectorRequirement
+	AttributeExpressions []metav1.LabelSelectorRequirement `json:"attributeExpressions,omitempty" protobuf:"bytes,2,rep,name=attributeExpressions"`
```

Comment:
Suggest we remove this for 0.4.0, to give us more time to determine requirements.

Reply:
Will do, let's close this PR then and leave the filtering as is for now.

Reply:
Let's at least get this level indirection in. `spec.filter` with Type and Source somehow inside there (whether directly or another nested field, is up to you).